src/ncp/ncp_base_radio.cpp

```diff
@@ -336,33 +336,58 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_MAC_15_4_SADDR>(void)
     return error;
 }
 
+otError NcpBase::DecodeStreamRawTxRequest(otRadioFrame &aFrame)
+{
+    otError        error;
+    const uint8_t *payloadPtr;
+    uint16_t       payloadLen;
+    bool           csmaEnable;
+
+    SuccessOrExit(error = mDecoder.ReadDataWithLen(payloadPtr, payloadLen));
+    VerifyOrExit(payloadLen <= OT_RADIO_FRAME_MAX_SIZE, error = OT_ERROR_PARSE);
+
+    aFrame.mLength = static_cast<uint8_t>(payloadLen);
+    memcpy(aFrame.mPsdu, payloadPtr, aFrame.mLength);
+
+    // Parse the meta data
+
+    // Channel is a required parameter in meta data.
+    SuccessOrExit(error = mDecoder.ReadUint8(aFrame.mChannel));
+
+    // Set the default value for all optional parameters.
+    aFrame.mInfo.mTxInfo.mMaxCsmaBackoffs = OPENTHREAD_CONFIG_MAC_MAX_CSMA_BACKOFFS_DIRECT;
+    aFrame.mInfo.mTxInfo.mMaxFrameRetries = OPENTHREAD_CONFIG_MAC_MAX_FRAME_RETRIES_DIRECT;
+    aFrame.mInfo.mTxInfo.mCsmaCaEnabled   = true;
+
+    // All the next parameters are optional. Note that even if the
+    // decoder fails to parse any of optional parameters we still want to
+    // return `OT_ERROR_NONE` (so `error` is not updated after this
+    // point).
+
+    SuccessOrExit(mDecoder.ReadUint8(aFrame.mInfo.mTxInfo.mMaxCsmaBackoffs));
+    SuccessOrExit(mDecoder.ReadUint8(aFrame.mInfo.mTxInfo.mMaxFrameRetries));
+    SuccessOrExit(mDecoder.ReadBool(csmaEnable));
+    aFrame.mInfo.mTxInfo.mCsmaCaEnabled = csmaEnable;
```

Comment:
I've been wondering if we could deduce this field. Is csma enabled if maxcsmabackoffs is non-zero?

Reply:
I don't think we can. 
-  If `mMaxCsmaBackoffs` is set to zero, radio should try a single tx attempt (and before tx started it should perform carrier-sense - assuming of course `mCsmaCaEnabled` is `true`). I think the `mMaxCsmaBackoffs` gives max number of backoffs between attempts (so zero means one attempt, one means two attempts, etc.)
-  `mCsmaCaEnabled` indicates whether radio should perform carrier-sense before tx or not.